### PR TITLE
First cut of the more general API

### PR DESCRIPTION
**Pull request by inaimathi**
Working edition. Kick the tires, and give it a head-check, but it should work.

### CODE REVIEW

File: src/knode/server.clj [hunk 1]
```diff
@@ -469,6 +473,49 @@
...
+                 (core/resolve-name (:env @state) {:label %})
+                 (catch Exception e nil)))
+             labels)))
+      (json-error 400 "Some given labels are not resolvable")
+
+      :else  (let [result (->> iris
+                               (map #(sparql/full-term @state % labels))
+                               (map #(into {} (map (fn [[k v]] [k (string/join " | " v)]) %))))]
+               {:status 200
+                :headers {"Content-Type" "application/json"}
```
Comment (jamesaoverton) [inline src/knode/server.clj]: should be "text/tab-separated-values"

**Comment by jamesaoverton** [comment]
I POST to `https://knode-dev.ontodev.com/api/get-terms?select=label`:

```
CURIE
EXAMPLE:0000001
```

and I get

```
CURIE	label
```

Then I try 

```
IRI
https://example.com/ontology/EXAMPLE_0000001
```

and I get 

```
IRI	label
	Example One
```

**Comment by inaimathi** [comment]
Fixed the above lookup issue. Note that the first one had two causes:

1. A bug wherein the wrong environment was being used
2. The lack of an `EXAMPLE` prefix in `ONTIE`.

If the `EXAMPLE` prefix is not present, the current return value is exactly what you've got noted above. what's the right thing to do in that situation?

**Comment by jamesaoverton** [comment]
I was running from a REPL with the environment from `test/example/` loaded, so EXAMPLE:0000001 did exist in `:terms` and the `:root-iri` was set properly (I think).

I'm not 100% sure, but I'm leaning toward including all requested IRIs in the results, even if they are not recognized by the system. This would match the behaviour in `get-term-status`.

**Comment by inaimathi** [comment]
(Just to clarify, yes, there was also a bug in the lookup logic, but it has been fixed at this point, 

```
~/projects/knode $ curl -X POST --data "CURIE
EXAMPLE:0000001" http://localhost:3210/api/get-terms?select=label
CURIE	label
EXAMPLE:0000001	Example One
~/projects/knode $ curl -X POST --data "IRI
https://example.com/ontology/EXAMPLE_0000001" http://localhost:3210/api/get-terms?select=label
IRI	label
https://example.com/ontology/EXAMPLE_0000001	Example One
~/projects/knode $
```

assuming that the appropriate labels/prefixes exist locally)

**Comment by jamesaoverton** [comment]
Closing this in favour of #30.

**Pull request closed without merging**